Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -5457,8 +5517,10 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parseUnaryExpress
     bool hasPrefixUpdateOp = false;
     unsigned lastOperator = 0;
 
-    if (UNLIKELY(match(AWAIT) && (currentFunctionScope()->isAsyncFunctionBoundary() || isModuleParseMode(sourceParseMode()))))
+    if (UNLIKELY(match(AWAIT) && (currentFunctionScope()->isAsyncFunctionBoundary() || isModuleParseMode(sourceParseMode())))) {
+        semanticFailIfTrue(!findClosetAsyncFunctionScope(), "Cannot use 'await' within static block scope");
```

Comment:
Looks like this is not checking static-block, right?

Reply:
Sorry, the function name `findClosetAsyncFunctionScope ` should be replaced with `findClosetAsyncFunctionScopeUntilStaticBlock`. Please see the examples with class static block, async function, and await above.